spec/models/document_type_selection_spec.rb

```diff
@@ -0,0 +1,68 @@
+require "json"
+
+RSpec.describe DocumentTypeSelection do
+  let(:document_type_selections) { YAML.load_file(Rails.root.join("config/document_type_selections.yml")) }
+
+  describe "all configured document types selections are valid" do
+    let(:document_type_selection_schema) { JSON.parse(File.read("config/schemas/document_type_selection.json")) }
+
+    it "conforms to the document type selection schema" do
+      document_type_selections.each do |document_type_selection|
+        validator = JSON::Validator.fully_validate(document_type_selection_schema, document_type_selection)
+        expect(validator).to(
+          be_empty,
+          "Validation for #{document_type_selection['id']} failed: \n\t#{validator.join("\n\t")}",
+        )
+      end
+    end
+
+    it "finds the corresponding object for every string id in the options" do
+      document_type_selections.flat_map { |d| d["options"] }.each do |option|
+        if option["type"] == "parent"
+          expect(described_class.find(option["id"]))
+            .to be_a(described_class)
+        end
+      end
```

Comment:
Rather than using the YAML file should we being using the DocumentTypeSelection class? and I also wonder if we can get this to be a single expectation rather than a loop (I guess this has about 35 expectations currently?

An idea would be 

```
all_options = described_class.all.flat_map(&:options)
expect { all_options.each { |option| described_class.find(option.id) } }
  .not_to raise_error(DocumentTypeSelection::NotFoundError)
```

Reply:
At the moment there should only be 4 expectations.

This test was written before the options were modelled, so `option.id` wouldn't have been available.

Reply:
Also we'd need to put a guard around the `expect` or pluck out the options of type `document_type_selection` because they are the only ones that should pass this test.

Reply:
It would work if the first line changed to:
```
all_options = described_class.all.flat_map(&:options).select(&:document_type_selection?)
```